src/simulators/statevector/qubitvector_thrust.hpp

```diff
@@ -622,6 +624,12 @@ cvector_t<data_t> QubitVectorThrust<data_t>::vector() const
   return ret;
 }
 
+template <typename data_t>
+cdict_t<data_t> QubitVectorThrust<data_t>::vector_ket(double epsilon) const{
+    // non-optimized version; relies on creating a copy of the statevector
+    return AER::Utils::vec2ket(vector(), epsilon, 16);
```

Comment:
I think this going to copy the full GPU vector to system memory before doing the conversion. It would be better if we could avoid this @doichanj

Reply:
Additional buffer will be required to make chopped vector on GPU (or on CPU) and I think it is not easy to parallelize making chopped vector. I think we can not avoid copying whole vector to CPU or accessing to whole vector from CPU to make chopped vector.